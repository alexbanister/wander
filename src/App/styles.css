* {
  box-sizing: border-box; }

body {
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-family: 'Quicksand', sans-serif;
  background: url("../assets/hero1.jpg");
  background-position: center;
  background-size: cover; }

a {
  text-decoration: none; }
  a:hover {
    color: #AEE173; }

.app {
  display: flex;
  flex-direction: row;
  width: 100vw;
  min-height: 100vh; }

main {
  width: 65vw;
  padding: 3em;
  background: rgba(64, 95, 115, 0.4); }

input, button, select {
  font-size: 1.25rem;
  padding: .5rem;
  height: 2.5rem; }
