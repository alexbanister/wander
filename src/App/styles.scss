@import "../variables.scss";

* {
  box-sizing: border-box;
}
body {
  margin: 0;
  padding: 0;
  color: $background;
  font-family: 'Quicksand', sans-serif;
  background: url('../assets/hero1.jpg');
  background-position: center;
  background-size: cover;
}
a {
  text-decoration: none;
  &:hover {
    color: $highlight-dark;
  }
}
.app {
  display: flex;
  flex-direction: row;
  width: 100vw;
  min-height: 100vh;
}
main {
  width: 65vw;
  overflow-y: scroll;
  height: 100vh;
  padding: 3em;
  background: $main-background;
}

input, button, select {
  font-size: 1.25rem;
  padding: .5rem;
  height: 2.5rem;
}
