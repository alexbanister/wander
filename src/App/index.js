import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Header from '../Header/';
import Welcome from '../Welcome/';
import User from '../User/';
import BucketList from '../BucketList/';
import Preferences from '../Preferences/';
import Search from '../Search/';
import './styles.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <Header />
        <main>
          <Route
            exact path='/'
            component={Welcome}
            useKey={false}
            key='Welcome'/>
          <Route
            exact path='/login'
            component={User}
            useKey={false}
            key='Login'/>
          <Route
            exact path='/signOut'
            render={()=> <User signOut={true} />}
            useKey={false}
            key='signOut'/>
          <Route
            exact path='/signup'
            render={()=> <User form='signup' />}
            useKey={false}
            key='signup'/>
          <Route
            exact path='/bucketList'
            component={BucketList}
            useKey={false}
            key='BucketList'/>
          <Route
            exact path='/preferences'
            component={Preferences}
            useKey={false}
            key='Preferences'/>
          <Route
            exact path='/search'
            component={Search}
            useKey={false}
            key='Search'/>
        </main>
      </div>
    );
  }
}

export default App;
