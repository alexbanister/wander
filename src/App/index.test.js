import React from 'react';
import ReactDOM from 'react-dom';
import App from './index';
import { shallow } from 'enzyme';

describe('App snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
