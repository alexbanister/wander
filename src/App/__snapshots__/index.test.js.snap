// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <main>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                    useKey={false}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
                    useKey={false}
          />
          <Route
                    exact={true}
                    path="/signOut"
                    render={[Function]}
                    useKey={false}
          />
          <Route
                    exact={true}
                    path="/signup"
                    render={[Function]}
                    useKey={false}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/bucketList"
                    useKey={false}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/preferences"
                    useKey={false}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/search"
                    useKey={false}
          />
</main>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
              useKey={false}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
              useKey={false}
/>,
            <Route
              exact={true}
              path="/signOut"
              render={[Function]}
              useKey={false}
/>,
            <Route
              exact={true}
              path="/signup"
              render={[Function]}
              useKey={false}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/bucketList"
              useKey={false}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/preferences"
              useKey={false}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/search"
              useKey={false}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Welcome",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Login",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "signOut",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signOut",
              "render": [Function],
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "signup",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signup",
              "render": [Function],
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "BucketList",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/bucketList",
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Preferences",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/preferences",
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Search",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/search",
              "useKey": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <main>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                        useKey={false}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/login"
                        useKey={false}
            />
            <Route
                        exact={true}
                        path="/signOut"
                        render={[Function]}
                        useKey={false}
            />
            <Route
                        exact={true}
                        path="/signup"
                        render={[Function]}
                        useKey={false}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/bucketList"
                        useKey={false}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/preferences"
                        useKey={false}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/search"
                        useKey={false}
            />
</main>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
                useKey={false}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
                useKey={false}
/>,
              <Route
                exact={true}
                path="/signOut"
                render={[Function]}
                useKey={false}
/>,
              <Route
                exact={true}
                path="/signup"
                render={[Function]}
                useKey={false}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/bucketList"
                useKey={false}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/preferences"
                useKey={false}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/search"
                useKey={false}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Welcome",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Login",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "signOut",
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/signOut",
                "render": [Function],
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "signup",
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/signup",
                "render": [Function],
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "BucketList",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/bucketList",
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Preferences",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/preferences",
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Search",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/search",
                "useKey": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
